test/conformance/kubetestgen/kubetestgen.go

```diff
@@ -0,0 +1,234 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/go-openapi/analysis"
+	"github.com/go-openapi/loads"
+	"github.com/go-openapi/spec"
+	"gopkg.in/yaml.v2"
+)
+
+type options struct {
+	schemaPath   string
+	resource     string
+	area         string
+	behaviorsDir string
+}
+
+func parseFlags() *options {
+	o := &options{}
+	flag.StringVar(&o.schemaPath, "schema", "", "Path to the OpenAPI schema")
+	flag.StringVar(&o.resource, "resource", ".*", "Resource name")
+	flag.StringVar(&o.area, "area", "default", "Area name to use")
+	flag.StringVar(&o.behaviorsDir, "dir", "../behaviors/", "Path to the behaviors directory")
+	flag.Parse()
+	return o
+}
+
+var defMap map[string]analysis.SchemaRef
+
+func main() {
+	defMap = make(map[string]analysis.SchemaRef)
+	o := parseFlags()
+
+	d, err := loads.JSONSpec(o.schemaPath)
+	if err != nil {
+		fmt.Printf("ERROR: %s\n", err.Error())
+		os.Exit(1)
+	}
+	defs := d.Analyzer.AllDefinitions()
+	sort.Slice(defs, func(i, j int) bool { return defs[i].Name < defs[j].Name })
+
+	for _, d := range defs {
+		if !d.TopLevel {
+			continue
+		}
+		defMap[d.Ref.String()] = d
+	}
+
+	var suites []Suite
+	var suiteMapping = make(map[string]*Suite)
+
+	for _, v := range defs {
+		if !v.TopLevel || o.resource != v.Name {
+			continue
+		}
+		name := trimObjectName(v.Name)
+
+		defaultsuite := Suite{
+			Suite:       o.area + "/spec",
+			Description: "Base suite for " + o.area,
+			Behaviors:   []Behavior{},
+		}
+
+		_ = defaultsuite
+
+		for p, propSchema := range v.Schema.Properties {
+			id := o.area + p + "/"
+
+			if propSchema.Ref.String() != "" || propSchema.Type[0] == "array" {
+				if _, ok := suiteMapping[id]; !ok {
+					newsuite := Suite{
+						Suite:       o.area + "/" + p,
+						Description: "Suite for " + o.area + "/" + p,
+						Behaviors:   []Behavior{},
+					}
+					suiteMapping[id] = &newsuite
+				}
+				behaviors := suiteMapping[id].Behaviors
+				behaviors = append(behaviors, schemaBehavior(o.area, name, p, propSchema)...)
+				suiteMapping[id].Behaviors = behaviors
+			} else {
+				if _, ok := suiteMapping["default"]; !ok {
+					newsuite := Suite{
+						Suite:       o.area + "/spec",
+						Description: "Base suite for " + o.area,
+						Behaviors:   []Behavior{},
+					}
+					suiteMapping["default"] = &newsuite
+				}
+
+				behaviors := suiteMapping["default"].Behaviors
+				behaviors = append(behaviors, schemaBehavior(o.area, name, p, propSchema)...)
+				suiteMapping["default"].Behaviors = behaviors
+
+			}
+		}
+		for _, v := range suiteMapping {
+			suites = append(suites, *v)
+		}
+
+		break
+	}
+
+	var area Area = Area{o.area, suites}
+	countFields(suites)
+	printYAML(o.behaviorsDir+o.area, area)
+}
+
+func printYAML(fileName string, areaO Area) {
+	f, err := os.Create(fileName + ".yaml")
+	if err != nil {
+		fmt.Printf("ERROR: %s\n", err.Error())
```

Comment:
It seems like we shouldn't proceed if we can't create a file. Also, in this case `f.Close()` and `WriteString()` will fail anyway. I've submitted PR to improve the error handling: https://github.com/kubernetes/kubernetes/pull/86874 PTAL